src/components/FlawFormOwner.vue

```diff
@@ -1,20 +1,32 @@
 <script setup lang="ts">
 import EditableText from '@/components/widgets/EditableText.vue';
 import { useUserStore } from '@/stores/UserStore';
-import { nextTick, computed } from 'vue';
+import { nextTick, computed, ref } from 'vue';
+
 const owner = defineModel<string | null>({ default: null });
 const userStore = useUserStore();
 
-function selfAssign() {
-  owner.value = userStore.userName;
+async function selfAssign() {
+  isLoading.value = true;
+  userStore.updateJiraUsername().then(() => {
```

Comment:
Return or await the promise chain to preserve the async behavior
```suggestion
  return userStore.updateJiraUsername().then(() => {
```